Examples/Algorithms/Fatras/src/NuclearInteractionOptions.cpp

```diff
@@ -0,0 +1,233 @@
+  // The collection
+  ActsFatras::detail::MultiParticleParametrisation mpp;
+
+  // Now read file
+  ActsFatras::detail::Parametrisation parametrisation;
+  TFile tf(fileName.c_str(), "read");
+  gDirectory->cd();
+  auto listOfParticles = gDirectory->GetListOfKeys();
+  auto initialParticle = listOfParticles->First();
+  while (initialParticle) {
+    // Get the initial particle
+    char const* particleName = initialParticle->GetName();
+    gDirectory->cd(particleName);
+
+    // Walk over all initial momenta for a particle
+    auto listOfMomenta = gDirectory->GetListOfKeys();
+    auto initialMomentum = listOfMomenta->First();
+    while (initialMomentum) {
+      // Parameters for a fixed inital momentum
+      ActsFatras::detail::Parameters parameters;
+      // Get the initial momentum
+      char const* nameMomentum = initialMomentum->GetName();
+      parameters.momentum = std::stof(nameMomentum);
+      gDirectory->cd(nameMomentum);
+
+      // Get the nuclear interaction probability
+      parameters.nuclearInteractionProbability = readHistogram(
+          "NuclearInteractionBinBorders", "NuclearInteractionBinContents");
+
+      // Get the soft interaction probability
+      TVectorF* softInteraction = (TVectorF*)gDirectory->Get("SoftInteraction");
+      parameters.softInteractionProbability = (*softInteraction)[0];
+
+      // Get the branching probabilities
+      std::vector<int> branchingPdgIds =
+          *((std::vector<int>*)gDirectory->Get("BranchingPdgIds"));
+      std::vector<int> targetPdgIds =
+          *((std::vector<int>*)gDirectory->Get("TargetPdgIds"));
+      std::vector<float> targetPdgProbability =
+          *((std::vector<float>*)gDirectory->Get("TargetPdgProbability"));
+      parameters.pdgMap.reserve(branchingPdgIds.size());
+      for (unsigned int i = 0; i < branchingPdgIds.size(); i++) {
+        auto it = parameters.pdgMap.begin();
+        while (it->first != branchingPdgIds[i] && it != parameters.pdgMap.end())
+          it++;
+
+        const auto target =
+            std::make_pair(targetPdgIds[i], targetPdgProbability[i]);
+        if (it != parameters.pdgMap.end()) {
+          it->second.push_back(target);
+        } else {
+          parameters.pdgMap.push_back(std::make_pair(
+              branchingPdgIds[i], std::vector<std::pair<int, float>>{target}));
+        }
+      }
+
+      // Get the soft distributions
+      gDirectory->cd("soft");
+      // Get the multiplicity distribution
+      parameters.softMultiplicity =
+          readHistogram("MultiplicityBinBorders", "MultiplicityBinContents");
+      // Get the distributions for each final state multiplicity
+      auto softList = gDirectory->GetListOfKeys();
+      auto softElement = softList->First();
+      while (softElement) {
+        readKinematicParameters(parameters, softElement, true);
+        softElement = softList->After(softElement);
+      }
+
+      // Get the hard distributions
+      gDirectory->cd("../hard");
```

Comment:
That's quite `../hard` coded.

Reply:
Since it reads the file generated by the `RootNuclearInteractionParametrisationWriter`, I guess one can do that